tracks/ruby/exercises/resistor-color-duo/mentoring.md

```diff
@@ -19,14 +23,30 @@ end
 
 ```
 
+### Reasonable Variants
+
+```ruby
+# with a hash instead of an array 
+COLOR_CODES.fetch_values(*colors).join.to_i
+
+# and &Parameter
+colors.map(&COLOR_CODES).join.to_i
+
+# or a hash with symbols
+colors.map { |color| COLOR_CODES[color.to_sym] }.join.to_i
+```
+
+- Using a class or a module (no reason to discuss). 
+- Multiplying the first value and adding the second one.     
+
 ### General
 
 This is the very first Array exercise, and the first loop. Introducing `map` is key in this exercise. 
 
-You, dear mentor, are totally right that using a hash would be a better solution, because it makes the relation between the color and its value explicit. But we really want people to know how to work with arrays before we ask them to work with hashes. (That will be taken care of later on in the track.) If a student _starts_ with a hash themselves, then by all means go with the hash.
+You, dear mentor, are totally right that using a hash would be a better solution, because it makes the relation between the color and its value explicit. But we really want people to know how to work with arrays before we ask them to work with hashes. (That will be taken care of later on in the track.) If a student _starts_ with a hash themselves, then by all means go with the hash. 
```

Comment:
```suggestion
You, dear mentor, are totally right that using a hash would be a better solution, because it makes the relation between the color and its value explicit. But we really want people to know how to work with arrays before we ask them to work with hashes. (That will be taken care of later on in the track.) If a student _starts_ with a hash themselves, then by all means go with the hash.
```